docs: Update github-runner-operator to Mature home page milestone (#801)
canonical/github-runner-operator: docs/index.md

Reviewer: Nit: I think we should remove these lines from the home page. Security is now explicitly surfaced in the table below, and some of this information is duplicated in previous parts of the home page. More specifically, these lines repeat that the runners are isolated in a single-user virtual machine, and these lines repeat that observability is available for this charm.

<!-- vale Canonical.007-Headings-sentence-case = NO -->

# GitHub runner operator

<!-- vale Canonical.007-Headings-sentence-case = YES -->

> A [Juju](https://juju.is/) [charm](https://documentation.ubuntu.com/juju/3.6/reference/charm/) for deploying and managing [GitHub self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) on virtual machines. The charm maintains a set of self-hosted runners, each isolated in a single-use virtual machine instance.

Reviewer: ```suggestion
A [Juju](https://juju.is/) [charm](https://canonical.com/juju/docs/juju-cli/3.6/reference/charm/) for deploying and managing [GitHub self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) on virtual machines. The charm maintains a set of self-hosted runners, each isolated in a single-use virtual machine instance.
```
nit, while we're here we should update the Juju docs link :)

| Get started | [Deploy the GitHub runner charm](local-lxd/tutorial/quick-start.md) • [Spawn OpenStack runner](how-to/openstack-runner.md)                                                                                                                                                                                                                           |
| Deployment  | [Run on LXD cloud](local-lxd/how-to/run-on-lxd.md) • [Change repository or organization](how-to/change-path.md) • [Change GitHub authentication](how-to/change-token.md)                                                                                                                                                                             |
| Operations  | [Configure runner storage](local-lxd/how-to/configure-runner-storage.md) • [Integrate with COS](how-to/integrate-with-cos.md) • [Debug with SSH](how-to/debug-with-ssh.md) • [Upgrade](how-to/upgrade.md)                                                                                                                                            |

> | LXD runners | [Add custom labels](local-lxd/how-to/add-custom-labels.md) • [Set base image](local-lxd/how-to/set-base-image.md)                                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| Deployment  | [Change repository or organization](how-to/change-path.md) • [Change GitHub authentication](how-to/change-token.md)                                                                                                                                                                             |
| Operations  | [Configure runner storage](local-lxd/how-to/configure-runner-storage.md) • [Integrate with COS](how-to/integrate-with-cos.md) • [Debug with SSH](how-to/debug-with-ssh.md) • [Upgrade](how-to/upgrade.md)                                                                                                                                            |
| LXD runners | [Run on LXD cloud](local-lxd/how-to/run-on-lxd.md) • [Add custom labels](local-lxd/how-to/add-custom-labels.md) • [Set base image](local-lxd/how-to/set-base-image.md)                                                                                                                                                                                                                                    |
```
I think we should consolidate as much of the docs related to the LXD runners into a single row.

Which leads me to a question: Is the "Configure runner storage" guide specific to the LXD runners, or would someone using OpenStack runners be able to configure runner storage using the same guide?

| Design      | [Charm architecture](reference/charm-architecture.md) • [Integrations](reference/integrations.md) • [ARM64](local-lxd/explanation/arm64.md)                                                                                                                                                                                                          |

> | Security    | [Security overview](explanation/security.md) • [Cryptographic overview](reference/cryptographic-overview.md) • [Token scopes](reference/token-scopes.md) • [External Access](reference/external-access.md) • [Comply with security requirements](how-to/comply-security.md) • [Manage external contributors](how-to/manage-external-contributors.md) |

Reviewer: ```suggestion
| Security    | [Security overview](explanation/security.md) • [Cryptographic overview](reference/cryptographic-overview.md) • [Token scopes](reference/token-scopes.md) • [External access](reference/external-access.md) • [Comply with security requirements](how-to/comply-security.md) • [Manage external contributors](how-to/manage-external-contributors.md) |
```
nit

| | |
|--|--|

> | Get started | [Quick start](local-lxd/tutorial/quick-start.md) • [Spawn OpenStack runner](how-to/openstack-runner.md) |

Reviewer: ```suggestion
| Get started | [Deploy the GitHub runner charm](local-lxd/tutorial/quick-start.md) • [Spawn OpenStack runner](how-to/openstack-runner.md) |
```
Nit, "quick start" is ambiguous about what the user will do

| Deployment | [Run on LXD cloud](local-lxd/how-to/run-on-lxd.md) • [Change repository or organization](how-to/change-path.md) • [Change GitHub authentication](how-to/change-token.md) |
| Operations | [Configure runner storage](local-lxd/how-to/configure-runner-storage.md) • [Integrate with COS](how-to/integrate-with-cos.md) • [Debug with SSH](how-to/debug-with-ssh.md) • [Upgrade](how-to/upgrade.md) |

> | LXD runners | [Add custom labels](local-lxd/how-to/add-custom-labels.md) • [Set base image](local-lxd/how-to/set-base-image.md) |

Reviewer: Question: Is there any documentation (reference or explanation) about the LXD runners, or the difference between LXD and OpenStack runners? I took a look through the documentation but didn't find anything.

Do you think it would benefit users to have this kind of document? (In a follow-up PR)

Author: There isn't quite a comparison doc, but at the same time we're moving to GARM charm in https://github.com/canonical/github-runner-operators/.
Perhaps it won't be worth the effort at this moment :)

| Design | [Charm architecture](reference/charm-architecture.md) • [Integrations](reference/integrations.md) • [ARM64](local-lxd/explanation/arm64.md) |
| Security | [Security overview](explanation/security.md) • [Cryptographic overview](reference/cryptographic-overview.md) • [Token scopes](reference/token-scopes.md) • [External Access](reference/external-access.md) • [Comply with security requirements](how-to/comply-security.md) • [Manage external contributors](how-to/manage-external-contributors.md) |

## How this documentation is organized

This documentation uses the [Diátaxis documentation structure](https://diataxis.fr/):

> * **Tutorials** — hands-on introductions for newcomers.

Reviewer 2: The Diátaxis orientation section lists "Tutorials — hands-on introductions for newcomers" as one of four categories, but the main (OpenStack) Contents TOC has no top-level Tutorial section — tutorials currently live only under "Track local-lxd". A newcomer reading this description may look for a main Tutorial section that isn't there.

Consider adding a brief parenthetical noting that hands-on tutorials currently live under the local-lxd track, or surfacing a top-level entry (e.g., a Quick start) in the main TOC to fulfil the expectation the description sets.

*🤖 AI-assisted*

Reviewer: Since the tutorial is geared toward the local-lxd track, I don't think we should move it to the main TOC. I agree that including a brief note here makes sense, maybe something like:
```suggestion
* **Tutorials** — hands-on introductions for newcomers using a local LXD controller.
```

| | |
|--|--|

> | Get started | [Managing resource usage](local-lxd/tutorial/managing-resource-usage.md) | [Quick start](local-lxd/tutorial/quick-start.md) |

Reviewer: Two comments here:
1. Quick start should come first in the list
2. "Managing resource usage" does not feel like a tutorial to me. It's not end-to-end, there are no commands for users to run, etc. If the content is still correct and relevant, I think we should move this document into the Explanation section.

> | Deployment | [Spawn OpenStack runner](how-to/openstack-runner.md) | [Run on LXD cloud](local-lxd/how-to/run-on-lxd.md) |

Reviewer: Other ideas for pages that we could include here: [Change repository or organization](https://github.com/canonical/github-runner-operator/blob/main/docs/how-to/change-path.md), [Change GitHub authentication](https://github.com/canonical/github-runner-operator/blob/main/docs/how-to/change-token.md)

I also think that [Configure runner storage](https://github.com/canonical/github-runner-operator/blob/main/docs/local-lxd/how-to/configure-runner-storage.md) might be a better fit in this row.

> | Operations | [Integrate with COS](how-to/integrate-with-cos.md) | [Debug with SSH](how-to/debug-with-ssh.md) | [Upgrade](how-to/upgrade.md) |

Reviewer: Other ideas for pages that we could include here: [Add custom labels](https://github.com/canonical/github-runner-operator/blob/main/docs/how-to/add-custom-labels.md), [Set base image](https://github.com/canonical/github-runner-operator/blob/main/docs/local-lxd/how-to/set-base-image.md), [Troubleshoot](https://github.com/canonical/github-runner-operator/blob/main/docs/how-to/troubleshoot.md)

But if you think any of these pages fall more into "Deployment" (meaning Day 0/1 operations), please let me know!

> | Product-specific feature | [Comply with security requirements](how-to/comply-security.md) | [Configure runner storage](local-lxd/how-to/configure-runner-storage.md) |

Reviewer: Two comments here:
1. "Product-specific feature" is a placeholder term -- we should update to something specific to the GH runner charm. 
2. I'm not sure what the product-specific feature you're trying to highlight with these pages. Maybe we should highlight the local-lxd track? Or perhaps the fact that LXD and OpenStack are both options? (I don't really know what "category" the LXD/OpenStack difference highlights -- maybe "runner types"? Let me know)

> | Design | [Charm architecture](reference/charm-architecture.md) | [Security](explanation/security.md) |

Reviewer: I think the Security page should go into the dedicated row

I also think we could add the [ARM64 explanation page](https://github.com/canonical/github-runner-operator/blob/main/docs/local-lxd/explanation/arm64.md) and maybe the [Integrations page](https://github.com/canonical/github-runner-operator/blob/main/docs/reference/integrations.md)

> | Security | [Cryptographic overview](reference/cryptographic-overview.md) | [Token scopes](reference/token-scopes.md) | [External Access](reference/external-access.md) |

Reviewer: The first thing I notice in the table's rendered output is that using the vertical lines `|` to separate the individual pages are causing issues in the table. So I think we should convert this to be `{list-table}` (see e.g. the [HAProxy home page](https://github.com/canonical/haproxy-operator/blob/main/docs/index.md?plain=1)). It won't render nicely until we set up an RTD project though. So we could also use `•` as separators instead.

> | Security | [Cryptographic overview](reference/cryptographic-overview.md) | [Token scopes](reference/token-scopes.md) | [External Access](reference/external-access.md) |

Reviewer: Other pages we could add to the Security row: [Security overview](https://github.com/canonical/github-runner-operator/blob/main/docs/explanation/security.md), [Comply with security requirements](https://github.com/canonical/github-runner-operator/blob/main/docs/how-to/comply-security.md), [Manage external contributors](https://github.com/canonical/github-runner-operator/blob/main/docs/how-to/manage-external-contributors.md)

## How this documentation is organized

This documentation uses the [Diátaxis documentation structure](https://diataxis.fr/):

* **Tutorials** — hands-on introductions for newcomers.
* **How-to guides** — step-by-step guides covering key operations and common tasks.
* **Reference** — technical information such as specifications, APIs, and configuration options.
* **Explanation** — background, concepts, and design discussion.

> * **Release notes** — summaries of changes and upgrade notes.

Reviewer: Since there are no release notes for the charm, I don't think we need to include it in this list

# Contents
<!-- vale Canonical.004-Canonical-product-names = NO -->
1. [How to](how-to)
   1. [Change repository or organization](how-to/change-path.md)
   1. [Change GitHub personal access token](how-to/change-token.md)
   1. [Add custom labels](how-to/add-custom-labels.md)
   1. [Debug with SSH](how-to/debug-with-ssh.md)
   1. [Integrate with COS](how-to/integrate-with-cos.md)
   1. [Spawn OpenStack runner](how-to/openstack-runner.md)
   1. [Comply with security requirements](how-to/comply-security.md)
   1. [Upgrade](how-to/upgrade.md)
   1. [Contribute](how-to/contribute.md)
1. [Reference](reference)
   1. [Actions](reference/actions.md)
   1. [Charm architecture](reference/charm-architecture.md)
   1. [Configurations](reference/configurations.md)
   1. [COS Integration](reference/cos.md)
   1. [GitHub runner cryptographic overview](reference/cryptographic-overview.md)
   1. [External Access](reference/external-access.md)
   1. [Integrations](reference/integrations.md)
   1. [Token scopes](reference/token-scopes.md)
1. [Explanation](explanation)
   1. [SSH Debug](explanation/ssh-debug.md)

>    1. [Managing resource usage](local-lxd/tutorial/managing-resource-usage.md)

Reviewer 2: `local-lxd/tutorial/managing-resource-usage.md` is listed under the top-level **Explanation** section of the Contents TOC, but its path places it in a `tutorial/` directory, and every other `local-lxd/*` entry appears under the "Track local-lxd" section further down. The Diátaxis orientation section added in this same PR explicitly distinguishes Tutorials from Explanation, so this placement sits at odds with both the file's path and the model just described.

Either move this entry under "Track local-lxd" → Explanation (alongside the other local-lxd content) so path and category agree, or relocate the source file out of `local-lxd/tutorial/` if classifying it as general explanation is the intent.

*🤖 AI-assisted*

Reviewer: Yes, please move `local-lxd/tutorial/managing-resource-usage.md` to `local-lxd/explanation/managing-resource-usage.md`. Thanks for catching, @yhaliaw !

   1. [Security](explanation/security.md)

> 1. [Track local-lxd](local-lxd)

Reviewer: Some general questions about the `local-lxd` track:
1. Is the long-term plan for this charm to support both LXD and OpenStack runners? If we're planning to drop support for one of the tracks, what would be the approximate time scale?
2. Do you think it would benefit readers if the local-lxd documentation had a landing page? (Meaning `docs/local-lxd/index.md`.) I think it might be good to have some formal introduction to this portion of the documentation.

Author: In the long term, we'll move to GARM https://github.com/cloudbase/garm, and it does have LXD provider: https://github.com/cloudbase/garm-provider-lxd/. 
I don't think we will continue with our local LXD deployment in this codebase, so probably it wouldn't benefit the users nor us in this case.

## Project and community

The GitHub runner operator is a member of the Ubuntu family. It's an open-source project that warmly welcomes community projects, contributions, suggestions, fixes, and constructive feedback.

### Governance and policies

* [Code of conduct](https://ubuntu.com/community/code-of-conduct)

### Get involved

* [Get support](https://discourse.charmhub.io/)
* [Join our online chat](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)
* [Contribute](https://github.com/canonical/github-runner-operator/blob/main/CONTRIBUTING.md)

### Releases

> * [Changelog](changelog.md)

Reviewer: this might be a little misleading since the changelog isn't really capturing releases -- I think we should remove

## Project and community

The GitHub runner operator is a member of the Ubuntu family. It's an open-source project that warmly welcomes community projects, contributions, suggestions, fixes, and constructive feedback.

### Governance and policies

- [Code of conduct](https://ubuntu.com/community/code-of-conduct)

### Get involved

> - [Get support](https://discourse.charmhub.io/)
> - [Join our online chat](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)
> - [Contribute](https://github.com/canonical/github-runner-operator/blob/main/CONTRIBUTING.md)
> 
> ### Releases
> 
> Thinking about using the GitHub runner operator for your next project? [Get in touch](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)!

Reviewer: ```suggestion
### Get involved

- [Get support](https://discourse.charmhub.io/)
- [Join our online chat](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)
- [Contribute](https://github.com/canonical/github-runner-operator/blob/main/CONTRIBUTING.md)

Thinking about using the GitHub runner operator for your next project? [Get in touch](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)!
```
no need for the "Releases" header at this time!
